vnext/Shared/DevSupportManager.cpp

```diff
@@ -47,7 +47,32 @@ std::future<std::pair<std::string, bool>> DownloadFromAsync(const std::string &u
   co_await winrt::resume_background();
 
   winrt::Windows::Web::Http::HttpRequestMessage request(winrt::Windows::Web::Http::HttpMethod::Get(), uri);
-  winrt::Windows::Web::Http::HttpResponseMessage response = co_await httpClient.SendRequestAsync(request);
+  auto asyncRequest = httpClient.SendRequestAsync(request);
+#ifdef DEFAULT_CPPWINRT_EXCEPTIONS
+  try {
+    winrt::Windows::Web::Http::HttpResponseMessage response = co_await asyncRequest;
+  } catch (winrt::hresult_error const &e) {
+    co_return std::make_pair(
+        Microsoft::Common::Unicode::Utf16ToUtf8(e.message().c_str(), e.message().size()).c_str(), false);
+  }
+#else
+  co_await lessthrow_await_adapter<winrt::Windows::Foundation::IAsyncOperationWithProgress<
+      winrt::Windows::Web::Http::HttpResponseMessage,
+      winrt::Windows::Web::Http::HttpProgress>>{asyncRequest};
+
+  HRESULT hr = asyncRequest.ErrorCode();
+  if (FAILED(hr)) {
+    std::ostringstream sstream;
+    if (hr == WININET_E_CANNOT_CONNECT) {
+      sstream << "A connection with the server " << url << " could not be established\n\nIs the packager running?";
+    } else {
+      sstream << "Error " << std::hex << static_cast<int>(asyncRequest.ErrorCode()) << " downloading " << url;
```

Comment:
@acoates-ms can we get the message in addition to the hresult? i.e. construct a hresult_error from it and then use the .message?

Reply:
winrt::hresult_error err(hr);
err.message() returns "The text associated with this error code could not be found." even on the most basic WININET_E_CANNOT_CONNECT hr that is returned here.

Reply:
@acoates-ms interesting, I can repro this too. It shouldn't happen. Filed: https://github.com/microsoft/cppwinrt/issues/710